authorization/v2/authorizer.go

```diff
@@ -92,8 +92,8 @@ func (a *authorizer) GetAuthorizedResources(token string, urisToCheck []UniformR
 	// filtering for resources the token has access to
 	var allowedResources []UniformResourceIdentifier
 	for _, resource := range claims.AllowedResources {
-		if resource.isSubsetOfAtLeastOne(urisToCheck) {
-			allowedResources = append(allowedResources, resource)
+		if match, matchedUri := resource.isSubsetOfAtLeastOne(urisToCheck); match {
+			allowedResources = append(allowedResources, matchedUri)
```

Comment:
TLDR: this change (and the one on `UniformResourceIdentifier.isSubsetOfAtLeastOne`) is not needed

The `GetAuthorizedURIs` operation needs to work a bit differently. It does not return URIs from the user's permissions, it returns the URIs from the request.

For example:
User has access to: `hs:hs_auth`
Request to `GetAuthorizedURIs` includes: `hs:hs_auth:api:v2:GetTeams`, `hs:hs_auth:api:v2:GetUsers`, `hs:hs_application:CreateApplication`
`GetAuthorizedURIs` would return: `hs:hs_auth:api:v2:GetTeams`, `hs:hs_auth:api:v2:GetUsers`, **not** `hs:hs_auth`

The idea here is that `hs_auth` clients would send an array of required URIs and then receive a subset of the sent array. So `GetAuthorizedURIs` is basically just a filter for what URIs the user does not have access to. If we returned URIs that encompass the URIs in the request rather than the URIs from the request, we'd need to implement additional logic on the clients for comparing if the returned URIs include the URIs in the request. When simply return the URIs `hs:hs_auth` received, we can then just do a simple string comparison on the client side to determine which URIs the user can access.

Reply:
Yeah, the `GetAuthorizedURIs` should return a subset of the URIs from the request. The change I made to the `isSubsetOfAtLeastOne` returns the URI from the slice that was provided to the function, I needed to do this since before we added the `resource` which is the URI from the token claims, rather than the URI from the request.

We iterate over the token claims, so if I instead iterate over the URIs provided in the request I can use the original version of `isSubsetOfAtLeastOne`.

Reply:
Riiight. It seems a bit weird to do it like that though. Couldn't you replace this loop with something like this?
```
for _, resource := range urisToCheck {
  if resource.isSubsetOfAtLeastOne(claims.AllowedResources) {
    allowedResources = append(allowedResources, resource)
  }
}
```

Reply:
I tried that, but due to the implementation if `isSubsetOf` it would fail using the `AllowedResources` as the function parameter. But looking at it, it seems like `resource.isSubsetOfAtLeastOne(claims.AllowedResources)` makes much more sense 🤔

Reply:
Hmm, the thing is, as far as I remember we're planning to remove the `AllowedResources` field from the tokens and just store the permissions for both user and service tokens on the DB. Should we maybe leave this PR open until we move the permissions to the database?

Reply:
Yeah, probably a good idea to wait for that change to be made